APMrover2/mode_auto.cpp

```diff
@@ -135,9 +139,16 @@ bool ModeAuto::reached_heading()
 // start RTL (within auto)
 void ModeAuto::start_RTL()
 {
-    if (_mode_rtl.enter()) {
-        _submode = Auto_RTL;
+    char reason[MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN] = {};
+    if (!_mode_rtl.ok_to_enter(reason, sizeof(reason))) {
```

Comment:
shouldn't this be sizeof(reason-1) as snprintf will append the \0 at the end ? otherwise we risk an overflow

Reply:
That's not typical of this sort of interface.